src/nvim/buffer.c

```diff
@@ -743,6 +743,8 @@ static void free_buffer(buf_T *buf)
   handle_unregister_buffer(buf);
   buf_free_count++;
   free_buffer_stuff(buf, true);
+  // b:changedtick uses an item in buf_T, remove it now.
+  tv_dict_item_remove(buf->b_vars, (dictitem_T *)&buf->changedtick_di);
```

Comment:
cc @ZyX-I

Reply:
Check `free_buffer_stuff`, it already removes changedtick. It is better to make `free_options` actually flag variable like the name already suggests and avoid calling `buf_init_changedtick()` based on some flag there: current workflow looks weird as you first remove `b:changedtick`, then readd it back by calling `buf_init_changedtick()`, then remove again.

Also based on my understanding it looks logical to not remove `b:changedtick`, but to *replace* it with something allocated if needed based on examining reference count before calling `unref_var_dict`. Because now you have key disappearing from dictionary for no good reason.